docs/aws/audit/dynamodbmonitoring/rules/sagemaker_notebook_data_encrypted_customer_master_keys_remediation.mdx

```diff
@@ -1,6 +1,228 @@
+   ```
+
+By following these steps, you can ensure that your DynamoDB table is encrypted using a KMS Customer Master Key (CMK), thereby preventing the misconfiguration of unencrypted notebook data.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the DynamoDB service by typing 'DynamoDB' in the search bar and selecting it from the dropdown menu.
+3. In the navigation pane on the left, click on 'Tables'.
+4. Select the table you want to inspect for encryption settings.
+5. In the 'Overview' tab, scroll down to the 'Table details' section. Here, you can see the 'Encryption' field. If it says 'KMS', then the table is encrypted with a KMS key. If it says 'Default', then it is not encrypted with a KMS key.
+6. To check which KMS key is being used, click on the 'Key ARN' link next to the 'Encryption' field. This will take you to the KMS console where you can see the details of the key being used for encryption.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the DynamoDB resources.
+
+2. Once the AWS CLI is set up, you can use the `list-tables` command to get a list of all the DynamoDB tables in your account. The command is as follows:
+
+   ```
+   aws dynamodb list-tables --region your-region
+   ```
+   Replace 'your-region' with the region your DynamoDB is located in.
+
+3. After getting the list of tables, you can use the `describe-table` command to get the details of each table. This command will provide you with the information about the KMS key used to encrypt the table. The command is as follows:
+
+   ```
+   aws dynamodb describe-table --table-name your-table-name --region your-region
+   ```
+   Replace 'your-table-name' with the name of your DynamoDB table and 'your-region' with the region your DynamoDB is located in.
+
+4. In the output of the `describe-table` command, look for the `SSEDescription` field. If the `SSEType` is `KMS` and the `KMSMasterKeyArn` is not empty, then the table is encrypted with a KMS Customer Master Key. If the `SSEType` is not `KMS` or the `KMSMasterKeyArn` is empty, then the table is not encrypted with a KMS Customer Master Key.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This will allow you to interact with AWS services including DynamoDB. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and establish a session with AWS using your access keys:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>